apps/client/src/pages/Dashboard.tsx

```diff
@@ -1,96 +1,94 @@
-    y: 9,
-    z: 1,
-    width: 28,
-    height: 15,
-    assets: [
-      {
-        assetId: DEMO_ASSET,
-        properties: [{ propertyId: DEMO_PROPERTY }],
-      },
-    ],
-  },
-  {
-    id: 'line-widget',
-    componentTag: 'iot-line-chart',
-    x: 31,
-    y: 7,
-    z: 1,
-    width: 46,
-    height: 35,
-    assets: [
-      {
-        assetId: DEMO_ASSET,
-        properties: [{ propertyId: DEMO_PROPERTY }],
-      },
-    ],
-  },
-];
+import { REGION } from '../testing/dashboardMocks';
 
 const Dashboard: React.FC = () => {
+  const dashboardsModule = new DashboardsModule();
   const matches = useMatches();
   const dashboardId = matches[matches.length - 1]?.params.id;
   const [query, setQuery] = useState<SiteWiseQuery>();
   const [client, setClient] = useState<IoTSiteWiseClient>();
+  const [dashboardConfiguration, setDashboardConfiguration] =
+    useState<DashboardConfiguration>({
+      widgets: [],
+      viewport: {
+        duration: '5m',
+      },
+    });
+  const [dashboardName, setDashboardName] = useState<string>();
+  const [editMode, setEditMode] = useState<boolean>(true);
 
-  useEffect(() => {
-    const setupCredentials = async () => {
-      const credentials = await Auth.currentCredentials();
+  const setupCredentials = async () => {
+    const credentials = await Auth.currentCredentials();
+    const region = REGION;
 
-      const { query: siteWiseQuery } = initialize({
-        awsCredentials: credentials,
-        awsRegion: region,
-      });
+    const { query: siteWiseQuery } = initialize({
+      awsCredentials: credentials,
+      awsRegion: region,
+    });
 
-      setQuery(siteWiseQuery);
+    setQuery(siteWiseQuery);
 
-      const siteWiseClient = new IoTSiteWiseClient({
-        region,
-        credentials,
-      });
+    const siteWiseClient = new IoTSiteWiseClient({
+      region,
+      credentials,
+    });
 
-      setClient(siteWiseClient);
-    };
+    setClient(siteWiseClient);
+  };
 
+  useEffect(() => {
     setupCredentials().catch((e) => {
       throw e;
     });
+
+    if (dashboardId) {
+      const fetchDashboard = async () => {
+        const d = await dashboardsModule.http.read(dashboardId);
+        setDashboardConfiguration(d.definition as DashboardConfiguration);
+        setDashboardName(d.name);
+      };
+
+      void fetchDashboard();
+    }
   }, []);
 
+  const saveDashboard = async (dashboard: DashboardConfiguration) => {
+    if (dashboardId && dashboardName) {
+      await dashboardsModule.http.update(dashboardId, dashboardName, dashboard);
+    }
+  };
+
   return (
     <React.Fragment>
-      <div>
-        <nav>
-          <Link to={'/dashboards'}>Dashboard list</Link>
-        </nav>
-        <h1>dashboard view for {dashboardId}</h1>
-      </div>
+      <Grid gridDefinition={[{ colspan: 11 }, { colspan: 1 }]}>
+        <Header>{dashboardName}</Header>
+        <Button onClick={() => setEditMode(!editMode)}>
+          {editMode ? 'view' : 'edit'}
```

Comment:
Intentionally not testing this yet since we will need to establish a pattern for integration tests and then I will backfill these. This button might end up being part of iot-app-kit dashboard instead.